parlai/core/teachers.py

```diff
@@ -480,6 +488,13 @@ def act(self):
         # TODO: all teachers should eventually create messages
         # while setting up the data, so this won't be necessary
         action = Message(action)
+
+        return action
+
+    def process_action(self, action: Message) -> Message:
```

Comment:
what is the purpose of adding these functions at such a high level? if it is important to keep, can we get a little more thorough documentation (why does this need to be split up into `get_orig_action` and `process_action`?

Reply:
The purpose of adding these functions at such a high level is because any wrapper teacher that modifies the 'text' or 'label' fields will have to do so before self.last_act or self.lastY are set, or otherwise the action will become out of sync with those attributes. (The alternative is to simply duplicate the logic of FixedDialogTeacher.act() in any wrapper teacher, but this will silently break whenever this method is changed in the future.) I've added a comment to the `.get_orig_action()` method to indicate why these methods are split up the way that they are; happy to discuss this more if need be

Reply:
so is it necessary for anyone who subclasses the `AbstractWrapperTeacher` to remember to call `get_orig_action` and then return `self.process_action`?

Reply:
It won't be necessary to do that manually - my plan is to refactor `AbstractWrapperTeacher` after this PR is in in order to call `get_orig_action()` and return `self.process_action()` in that abstract teacher's `.act()` directly. I plan to do this because I expect that all subclassed wrapper teachers will likely need to add their act-modifying code in between those two methods

Reply:
Ahh yes great, that would make this less prone to user error